src/Std/Internal/Async/Select.lean

```diff
@@ -0,0 +1,159 @@
+/-
+Copyright (c) 2025 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Henrik Böving
+-/
+prelude
+import Init.Data.Array.Basic
+import Init.Data.Random
+import Std.Internal.Async.Basic
+
+/-!
+This module contains the implementation of a fair and data-loss free IO multiplexing primitive.
+The main entrypoint for users is `Selectable.one` and the various functions to produces
+`Selector`s from other modules.
+-/
+
+namespace Std
+namespace Internal
+namespace IO
+namespace Async
+
+/--
+The core data structure for racing on winning a `Selectable.one` if multiple event sources are ready
+at the same time. A `Task` can try to finish the waiter by calling `Waiter.race`.
+-/
+structure Waiter (α : Type) where
+  private mk ::
+    private finished : IO.Ref Bool
+    promise : IO.Promise (Except IO.Error α)
+
+/--
+Create a fresh `Waiter`.
+-/
+def Waiter.new : BaseIO (Waiter α) := do
```

Comment:
I think I don't understand the purpose of this function. Is there a reason for anyone except for `Selectable.one` to create new `Waiter`s?

Similar question for `withPromise`.

Reply:
For `Waiter.new` no. For `withPromise` there is going to be a caller in a later PR though I can omit it for now if that's better?